lib/data/data_source/remote/database_remote_source_impl.dart

```diff
@@ -96,4 +99,30 @@ class DatabaseRemoteSourceImpl implements DatabaseRemoteSource {
       throw ApiException(Errors.somethingWentWrong);
     }
   }
+  @override
+  Future<Success> updateUserProfile(UserProfileDto dto) async {
+    try {
+      final userId = firebaseAuth.currentUser?.uid;
+      if (userId != null) {
+        final doc = await firestore.collection("users").doc(userId).get();
+        if (doc.exists) {
+          final UserProfileDto existingUserDoc = UserProfileDto.fromJson(doc.data()!);
+          final updatedUserDoc = existingUserDoc.copyWith(
+            name: dto.name.isNotEmpty ? dto.name : existingUserDoc.name,
+            userId: dto.userId.isNotEmpty ? dto.userId : existingUserDoc.userId,
+            initialLanguage: dto.initialLanguage.isNotEmpty ? dto.initialLanguage : existingUserDoc.initialLanguage,
+            userFolders: dto.userFolders.isNotEmpty ? dto.userFolders : existingUserDoc.userFolders,
+          );
+          await firestore.collection("users").doc(userId).set(updatedUserDoc.toJson());
+        } else {
+          await firestore.collection("users").doc(userId).set(dto.toJson());
+        }
+        return const Success();
+      } else {
+        throw ApiException(Errors.userNotFound);
+      }
+    } catch (e) {
+      throw ApiException(Errors.somethingWentWrong);
+    }
+  }
```

Comment:
This is database_remote_source where we keep methods for managing folders, words etc. The class name is a bit misleading, it should be more like `FoldersDataSource` or `UserFoldersDataSource`. However, the method for updating user profile should be placed at `AuthenticationDataSource`